### DISCUSSION TITLE
tkinter wouldn't work on code-server （remote）

### DISCUSSION THREAD

**Discussion by wabbyy** [coder/code-server#4561] [Ideas]
Will this function be added in further version？

----------------------------------------------------------
# /usr/local/bin/python3 /root/AJ-Python/temp/2.py
Traceback (most recent call last):
  File "/root/AJ-Python/temp/2.py", line 1, in <module>
    import turtle
  File "/usr/local/lib/python3.10/turtle.py", line 107, in <module>
    import tkinter as TK
  File "/usr/local/lib/python3.10/tkinter/__init__.py", line 37, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
ModuleNotFoundError: No module named '_tkinter'

**Comment by lg-nineoneninethreenine** [comment]
I expect you have a remote instance(local ones won't have a big problem since you could just `code-server &`, sending it into the background and using X forwarding for the display. :wink: trick)
The remote instance doesn't have a GUI. IF you have a local machine with Xorg(the most preferred windowing system in the Linux world), you are in luck. But on a server, it's hard. You'd end up getting a lot of useless software and finally choosing noVNC. Here's my suggestion. Install Xorg onto your machine. Get Xfce or LXDE. After that, install noVNC(people say it's `sudo apt update && sudo apt install novnc`) and visit port 6080(hope you know how to deal with that).
Maybe you could get to see Tkinter work on code-server. If it's a Docker deployment(expected), add the respective lines to the Dockerfile, make sure it _doesn't_ break(happened a LOT of times).
After that, you can upgrade code-server and enjoy the GUI.
